src/boot/efi/measure.c

```diff
@@ -42,6 +42,51 @@ static EFI_STATUS tpm1_measure_to_pcr_and_event_log(
                         &event_log_last);
 }
 
+static EFI_STATUS tpm2_measure_to_pcr_and_tagged_event_log(
+                EFI_TCG2_PROTOCOL *tcg,
+                uint32_t pcrindex,
+                EFI_PHYSICAL_ADDRESS buffer,
+                uint64_t buffer_size,
+                uint32_t event_id,
+                const char16_t *description) {
+
+        _cleanup_free_ struct event {
+                EFI_TCG2_EVENT tcg_event;
+                EFI_TCG2_TAGGED_EVENT tcg_tagged_event;
+        } *event;
+        size_t desc_len, event_size;
+
+        assert(tcg);
+        assert(description);
+
+        desc_len = strsize16(description);
+        event_size = offsetof(EFI_TCG2_EVENT, Event) + offsetof(EFI_TCG2_TAGGED_EVENT, Event) + desc_len;
+
+        event = xmalloc(event_size);
+
+        memset(event, 0, event_size);
```

Comment:
pleas rebase on https://github.com/systemd/systemd/pull/29405 and use xmalloc0() here

Reply:
#29405 has been merged now